design/nfs/nfs-quota.md

```diff
@@ -0,0 +1,68 @@
+# NFS Quota
+
+## Background
+
+Currently, when the user creates PersistentVolumes from an NFS Rook share/export via PersistentVolumeClaim, the provisioner does not provide the spesific capacity as requested. For example the users create PersistentVolumeClaim as following:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: rook-nfs-pv-claim
+spec:
+  storageClassName: "rook-nfs-share"
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Mi
+```
+
+The client still can use the higher capacity than `1mi` as requested.
+
+This proposal is to add features which the Rook NFS Provisioner will provide the specific capacity as requested from `.spec.resources.requests.storage` field in PersistentVolumeClaim.
+
+## Implementation
+
+The implementation will be use `Project Quota` on xfs filesystem. When the users need to use the quota feature they should use xfs filesystem with `prjquota/pquota` mount options for underlying volume. Users can specify filesystem type and mount options through StorageClass that will be used for underlying volume. For example:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: standard-xfs
+parameters:
+  fsType: xfs
+mountOptions:
+  - prjquota
+...
+```
+
+Then the underlying PersistentVolumeClaim should be used that StorageClass 
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: nfs-default-claim
+spec:
+  storageClassName: "standard-xfs"
+  accessModes:
+  - ReadWriteOnce
+...
+```
+
+If the above conditions are met then the Rook NFS Provisioner will create projects and set the quota limit using [xfs_quota](https://linux.die.net/man/8/xfs_quota) before creating PersistentVolumes based on `.spec.resources.requests.storage` field in PersistentVolumeClaim. Otherwise the Rook NFS Provisioner will provision a PersistentVolumes without creating setting the quota.
```

Comment:
Yes, they are different. each PVC which backed by NFS export will be a different `projects`. let's say PVC A will have projects id 1 and PVC B will have projects id 2 and so on. And each project has different size/capacity based on `.spec.resources.requests.storage`.

The mapping for PVC and `projects` will look like
|PVC|project entries|
|-|-|
|A|1:/pv-a-path:size|
|B|2:/pv-b-path:size|

The nfs-provisioner will also create a `projects` file for saving those project entries, so it will be persistent in case nfs-provisioner restarted/killed then nfs-provisioner can restore the state of quota for each directory (PV which backed by NFS export).

Reply:
Thank you for clarifying.  where would this `projects` file be persisted to exactly?